bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200HandlerFactory.java

```diff
@@ -57,12 +60,12 @@
 
     private final Logger logger = LoggerFactory.getLogger(KM200HandlerFactory.class);
 
-    private KM200ChannelTypeProvider channelTypeProvider;
+    private @Nullable KM200ChannelTypeProvider channelTypeProvider;
 
     /**
      * shared instance of HTTP client for asynchronous calls
      */
-    private HttpClient httpClient;
+    private @Nullable HttpClient httpClient;
```

Comment:
Do you maybe have an example you to do this? I'm not sure about the interface to the constructor of `KM200HandlerFactory` instead of the `reference` anotation to set/unset. How can I modify the call to the modified constructor? Usually the system is searching fo a constructor with 0 arguments. Is there a different annotation type for this case? Sorry, I don't know how to do this..

Reply:
```
@Activate
public KM200HandlerFactory(@Reference HttpClientFactory httpClientFactory){
...
}
```

Reply:
thx, ok, done.